microservices/download/utils.js

```diff
@@ -111,175 +107,193 @@ async function displayError(req, res, errMsg) {
 
 async function findLink(os, type) {
   try {
-    let repositoryQuery = `
-      query getRepositoryBuildStatuses {
-        repository(id: 6483909499158528) {
-          builds(branch: "master", last: 10) {
-            edges {
-              node {
-                id
-                status
-              }
-            }
-          }
+
+    let releases = await doRequest();
+
+    // Now these releases should be sorted already, if we find they aren't we might
+    // have to add semver as a dep on this microservice, which is no fun since this
+    // microservice has 0 deps currently. For now lets assume it's a sorted array
+    // This same assumption is made on the `pulsar-updater` core package
+
+    for (const version of releases) {
+      for (const asset of version.assets) {
+
+        let name = asset.name;
```

Comment:
```suggestion
        let name = asset?.name;
```

_If_ you want to put optional chaining back, I might have been misinformed about just how "impossible" it is for `asset` to be `undefined` here (I have found out it is indeed possible).

We're pulling data from JSON where `undefined` isn't allowed, but in general I think you had the motivation/implementation probably right before, sorry about that.